cmd/config/internal/commands/sink_test.go

```diff
@@ -136,6 +136,46 @@ spec:
 	}
 }
 
+func TestSinkCommandJSON(t *testing.T) {
+	d, err := ioutil.TempDir("", "kustomize-source-test")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	defer os.RemoveAll(d)
+
+	r := commands.GetSinkRunner("")
+	r.Command.SetIn(bytes.NewBufferString(`apiVersion: config.kubernetes.io/v1alpha1
+kind: ResourceList
+items:
+- kind: Deployment
+  metadata:
+    labels:
+      app: nginx2
+    name: foo
+    annotations:
+      app: nginx2
+      config.kubernetes.io/index: '0'
+      config.kubernetes.io/path: 'f1.json'
+      config.kubernetes.io/format: 'json'
+  spec:
+    replicas: 1
+`))
+	r.Command.SetArgs([]string{d})
+	if !assert.NoError(t, r.Command.Execute()) {
+		t.FailNow()
+	}
+
+	actual, err := ioutil.ReadFile(filepath.Join(d, "f1.json"))
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	expected := `{"kind":"Deployment","metadata":{"annotations":{"app":"nginx2"},"labels":{"app":"nginx2"},"name":"foo"},"spec":{"replicas":1}}
```

Comment:
We probably want to output to be multiline.

Reply:
Done.